### PR TITLE
Updated pre python 2.7 code

### LINKED ISSUE #360: Remove Python <2.7 compatibility fixes
There's at least a few places in the code with comments that refer to pre-2.7 compatibility. We still want to keep at Python 2.7 compatibility for now, but we can update anything that was put in for Python 2.6 or earlier compatibility.

### PR DESCRIPTION
**Pull request by Will-Cross1**
Mentions of using python versions before 2.7 have been updated to utilise features that are available in python 2.7+

Mentions of only python 2 have been ignored as 2.7 is included under it

Resolves #360

### CODE REVIEW

**Review by tofu-rocketry — CHANGES_REQUESTED** [review]

File: constraints.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Constraints that apply to pip installed requirements
 
 # coveralls dependencies that need older versions for Python 2.6
```
[the remark this answers is not in the document]
Comment (Will-Cross1) [reply constraints.txt]: Resolved in #fcbd42a

File: bin/parser.py [hunk 2]
```diff
@@ -106,22 +106,18 @@ def parse_file(parser, apel_db, fp, replace):
...
     index = 0
     parsed = 0
     failed = 0
     ignored = 0
 
-    for index, line in enumerate(fp):
-        # Indexing is from zero so add 1 to find line number. Can't use start=1
-        # in enumerate() to maintain Python 2.4 compatibility.
-        line_number = index + 1
+    for index, line in enumerate(fp, start=1):
```
[the remark this answers is not in the document]
Comment (Will-Cross1) [reply bin/parser.py]: Resolved in #fcbd42a

**Review by tofu-rocketry — APPROVED** [review]
Looks good.

**Comment by tofu-rocketry** [comment]
Good commit messages too.

**Pull request merged**